sdk/openai/Azure.AI.OpenAI/src/Custom/ChatCompletions/ChatCompletionsJsonResponseFormat.cs

```diff
@@ -0,0 +1,11 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#nullable disable
+
+namespace Azure.AI.OpenAI;
+
+// CUSTOM CODE NOTE: demoting visibility to internal as we'd like to make this an enum-like representation on the parent.
+
+internal partial class ChatCompletionsJsonResponseFormat : ChatCompletionsResponseFormat
```

Comment:
We should explore doing this change in the TypeSpec too. It should be possible to hide these types using the `usage` and/or `access` decorators.

Reply:
Agreed; let's follow on that together with some of the type representations for for request messages and the like. I bet we can do things better without deviating too much from spec.